src/reducers/booksSlice.js

```diff
@@ -0,0 +1,88 @@
+import {
+  createSlice,
+  createAsyncThunk,
+  createEntityAdapter,
+} from '@reduxjs/toolkit'
+import {client} from 'utils/api-client'
+
+const booksAdapter = createEntityAdapter()
+
+const initialState = booksAdapter.getInitialState({
+  status: 'idle',
+  error: null,
+})
+
+export const fetchBooksByQuery = createAsyncThunk(
+  'BOOKS/FETCH_BOOKS_BY_QUERY',
+  async (query, {getState}) => {
+    const token = getState().auth.user?.token
+    const data = await client(`books?query=${encodeURIComponent(query)}`, {
+      token,
+    })
+
+    return data.books
+  },
+)
+
+export const fetchBookById = createAsyncThunk(
+  'BOOKS/FETCH_BOOK_BY_ID',
+  async (bookId, {getState}) => {
+    if (getState().books.ids.includes(bookId)) {
+      return getState().books.entities[bookId]
+    }
+
+    const token = getState().auth.user?.token
```

Comment:
Should declare a variable state = getState() here to avoid calling the same function multiple times in one blcok